engine/engine.go

```diff
@@ -164,7 +163,7 @@ func (e *engine) Run() {
 
 	go e.runPoller()
 
-	e.runWebhookListener()
+	e.WebhookListener.Run()
```

Comment:
I think it would be best if it would be the "owner" of WebhookListener who calls `Run()`. Previously the listener was owned by engine so it made sense to do it here, but that is no longer the case. So I would move this to `main.go`, probably at the very end to make sure all routes have been registered by the time `Run()` is called.

Since WebhookListener is being injected into engine (i.e. it's no longer "owned" by engine), I think it shouldn't be engine who calls `Run()`.

Reply:
(Sorry, I didn't mean to add as single comment, still reviewing)

Reply:
Why do you think it's better for the "owner" to call `Run()`? When dependency has been injected (car keys given to somebody), they should have full control over what to do and when to do, as per they see it fit (when to start/stop engine or start moving).

Reply:
Great metaphor, I didn't realise it first, that we need multiple drivers. Even regardless of that, engine shouldn't get to dictate when the server would run.

I made the change, but it feels weird in terms of `engine.Run()` being non-blocking and its the webhook server that's blocking and holding the main thread.